exonum/src/runtime/rust/mod.rs

```diff
@@ -288,20 +289,56 @@ mod stubs;
 #[cfg(test)]
 mod tests;
 
+trait FactoryWithMigrations: ServiceFactory + MigrateData {}
+impl<T: ServiceFactory + MigrateData> FactoryWithMigrations for T {}
+
+/// Wrapper around a service factory that does not support migrations.
+#[derive(Debug)]
+struct WithoutMigrations<T>(T);
```

Comment:
Can user one day change their mind and switch to `WithMigrations`? If not, I'd suggest to add a big warning like "This cannot be undone! Think twice, think thrice and consult your psychotherapist before doing that." just to avoid tears and blames later :)

Reply:
What do you mean by switching to `WithMigrations`? If we consider recompiling the node binary acceptable, then sure, the user should just switch `with_factory` to `with_migrating_factory` in the Rust runtime builder and replace the node binary.

Reply:
Yeah, that exactly what I meant.

Then I think it will be nice to have a guide-level explanation of how to make your non-migration (at the moment) service support migration, e.g. "Add a new version of artifact, add a factory for it to the rust runtime, create migration script, change invoking of `with_factory` for artifact from which you want to migrate to `with_migrating_factory`, etc".

Reply:
This makes sense, but IMO this should be documented once we have workflow for the dispatcher / supervisor.